### PR TITLE
fix(cli): remove `ignore` from crawling, and fix regression

### LINKED ISSUE #1247: 🐛 Biome doesn't throw errors for files that can't handle
### Environment information

```block
main
```

### What happened?

Try to handle a file that Biome doesn't know how to handle.

I should print a diagnostic saying that it doesn't know how to handle a file. This is a regression that was introduced way back, and we didn't catch because we don't have a test for it

### Expected result

I should throw a diagnostic. Users should use `files.ignore` or `files.ignoreUnknown`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

Closes #1247 

The issue was introduced a long time ago (before the fork) and it went unnoticed by the majority of the contributors. 

This caused some weird issues and strange things from my end. I hunted down the issue and created some tests to avoid regressions.

Unfortunately, now that Biome gained some usage, we can't emit errors, otherwise, it will break many CIs once we release this fix. Because of that, I decided to emit a warning instead. 

Also, I decided to remove the `ignore` crate for crawling the file system. Unfortunately, it doesn't work very well for our needs, and it gets in the way of how we decide to handle symbolic links. Hence, I think we should handle nested `.gitignore` using a different strategy.

I removed the feature from the changelog.

## Test Plan

Added tests.

I did various tests by playing with the `files.ignore`, and making sure I got the correct diagnostics for the files Biome can't handle.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -120,7 +120,8 @@ bitflags          = "2.3.1"
 bpaf              = { version = "0.9.5", features = ["derive"] }
 countme           = "3.0.1"
 dashmap           = "5.4.0"
-ignore            = "0.4.20"
+globset           = "0.4.14"
+ignore            = "0.4.21"
```
Comment (Conaclos) [inline Cargo.toml:124]: DO we still need the `ignore` crate?

File: biome.json [hunk 2]
```diff
@@ -20,7 +20,8 @@
 			"*.scss",
 			"*.svg",
 			"*.png",
-			"*.yml"
+			"*.yml",
+			"**/undefined/**"
```
Comment (unvalley) [inline biome.json:24]: nits: Will this undefined dir be created by some project?

File: Cargo.toml (hunk 1 above)
Comment (ematipico) [reply Cargo.toml:124]: Yes, we use it to parse the `.gitignore` contents when we enable the VCS integration. `ignore` also has a [`::gitignore`](https://docs.rs/ignore/latest/ignore/gitignore/index.html) module that does the parsing and exposes utility functions.

File: biome.json (hunk 2 above)
Comment (ematipico) [reply biome.json:24]: This morning I found that our formatter goes into stack overflow when parsing this file: https://github.com/biomejs/biome/blob/main/crates/biome_json_parser/tests/json_test_suite/undefined/structure_500_nested_arrays.json

So I decided to ignore the folder for the time been. This occurs if you decide to run Biome to the entire repository..

More context: https://discord.com/channels/1132231889290285117/1132602492987908166/1186243771868921927

Comment (unvalley) [reply biome.json:24]: I got it, thank you!

File: website/src/pages/.editorconfig
Comment (ematipico) [inline website/src/pages/.editorconfig:1]: I removed it because the top `.editorconfig` already does the same

**Pull request merged**